### DISCUSSION TITLE
How to use an entity for trigger: state node conditions?

### DISCUSSION THREAD

**Discussion by Rickerdo** [zachowj/node-red-contrib-home-assistant-websocket#630] [Q&A] [answered]
I am trying to compare the state of the triggered entity with the state of another entity. The conditional entity never pulls the value of the entity I have listed. Perhaps my syntax is wrong, but there is no documentation to describe such a scenario.

In the screenshot below, a change in state correctly triggers the node. The node then attempts to compare the trigger state against the entity "input_number.master_bedroom_tstat_cool_pm" which is just a number input set to the number 70. It seems to me that the entity "input_number.master_bedroom_tstat_cool_pm" is not being parsed as an entity, but rather as a string because the error message never shows the value of the entity's state (see second screenshot below).

![image](https://user-images.githubusercontent.com/6199259/164010989-55d23262-7b7a-4b85-acb4-fac0029f36a1.png)

![image](https://user-images.githubusercontent.com/6199259/164011884-7b2f46ba-d047-4ee1-a966-eb4bda8f18aa.png)

**Comment by zachowj** [comment, marked as answer]
`entity.` references the entity that triggered the node, `sensor.master_bedroom_thermostat_air_temperature`.

If you want to check a different entity swap the condition.

![image](https://user-images.githubusercontent.com/37859597/164113348-8fe05eb6-2a1c-4a20-a9ec-3336192fde8c.png)

or use JSONata - `$entities("input_number.master_bedroom_tstat_cool_pm").state`

![image](https://user-images.githubusercontent.com/37859597/164113413-84ad528f-b668-4160-8e6b-52e6ee3011f9.png)


edit: Those should have all been using `input_number.master_bedroom_tstat_cool_pm`

**Reply by Rickerdo** [reply]
Although the examples you provided have the trigger entity listed in the condition (opposite of what I need), the syntax is what I needed. The following screenshot yields my expected results:

![image](https://user-images.githubusercontent.com/6199259/164115335-70747385-f321-4e17-9e32-0d0213f32d0b.png)

However, setting "entity id" to "input_number.master_bedroom_tstat_cool_pm" as you listed in the first screenshot, in the Conditions section, caused all sorts of odd results that I couldn't figure out a pattern for. Nevertheless, like I stated above, the syntax you provided in your second screenshot was exactly what I needed.

Could the docs for the "trigger: state" node be updated to show the JSONata expression: $entities("{entity}").state?

Thanks for the help and the quick response!
